XMonad/Util/Loggers.hs

```diff
@@ -182,6 +186,70 @@ logLayout = withWindowSet $ return . Just . ld
 logCurrent :: Logger
 logCurrent = withWindowSet $ return . Just . W.currentTag
 
+-- | Log the given string, as is.
+logConst :: String -> Logger
+logConst = return . Just
+
+-- | If the first logger returns @Nothing@, the default logger is used.
+-- For example, to display a quote when no windows are on the screen,
+-- you can do:
+--
+-- > logDefault logTitle (logConst "Hey, you, you're finally awake.")
+logDefault :: Logger -> Logger -> Logger
+logDefault l d = l >>= maybe d logConst
+
+-- | An infix operator for 'logMaybe', which can be more convenient to
+-- combine multiple loggers.
+--
+-- > logTitle .| logWhenActive 0 (logConst "*") .| logConst "There's nothing here"
+(.|) :: Logger -> Logger -> Logger
+(.|) = logDefault
+
+-- $xmonad-screen
+-- It is also possible to bind loggers like 'logTitle' to a specific screen. For
+-- example, using @logTitleOnScreen 1@ will log the title of the focused window
+-- on screen 1, even if screen 1 is not currently active.
+
+-- | Only display the 'Logger' if the screen with the given 'ScreenId' is
+-- active.
+-- For example, this can be used to create a marker that is only displayed
+-- when the primary screen is active.
+--
+-- > logWhenActive 0 (logConst "*")
+logWhenActive :: ScreenId -> Logger -> Logger
+logWhenActive n l = do
+  c <- withWindowSet $ return . W.screen . W.current
+  if n == c then l else return Nothing
+
+-- | Get the title (name) of the focused window, on the given screen.
+logTitleOnScreen :: ScreenId -> Logger
+logTitleOnScreen =
+  withScreen
+    $ traverse (fmap show . getName)
+    . (W.focus <$>)
+    . W.stack
+    . W.workspace
+
+-- | Get the name of the visible workspace on the given screen.
+logCurrentOnScreen :: ScreenId -> Logger
+logCurrentOnScreen = withScreen $ logConst . W.tag . W.workspace
+
+-- | Get the name of the visible layout on the given screen.
```

Comment:
Funnily enough, while "current workspace" -> "visible workspace" felt very natural to me, "current layout" -> "visible layout" feels wrong.  A layout is not really "visible" per se, is it?

Reply:
I'd agree. "Layout of the workspace visible on the given screen" is a mouthful but consistent with how we normally refer to such things.

Reply:
I guess it does, I updated it